client/clientfactory.go

```diff
@@ -217,12 +218,15 @@ func newServiceKeyResolver(resolver membership.ServiceResolver) *serviceKeyResol
 	}
 }
 
-func (r *serviceKeyResolverImpl) Lookup(key string) (string, error) {
-	host, err := r.resolver.Lookup(key)
-	if err != nil {
-		return "", err
+func (r *serviceKeyResolverImpl) Lookup(key string, n int) (string, error) {
+	hosts := r.resolver.LookupN(key, n+1)
+	if len(hosts) == 0 {
+		return "", membership.ErrInsufficientHosts
+	}
+	if n >= len(hosts) {
+		n %= len(hosts)
 	}
```

Comment:
This part could lead to inconsistent hashing when on host goes down/up. Is the assumption that n is comfortably lower than # of host?